tests/monitoring/monitoring.go

```diff
@@ -124,21 +126,9 @@ var _ = Describe("[Serial][sig-monitoring]Monitoring", Serial, decorators.SigMon
 			originalKv := util.GetCurrentKv(virtClient)
 			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(context.Background(), originalKv.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
-			kv.Spec.CustomizeComponents = v1.CustomizeComponents{
-				Patches: []v1.CustomizeComponentsPatch{
-					{
-						ResourceName: virtHandler.deploymentName,
-						ResourceType: "DaemonSet",
-						Patch:        `{"spec":{"template":{"spec":{"nodeSelector":{"kubernetes.io/hostname":"does-not-exist"}}}}}`,
-						Type:         v1.StrategicMergePatchType,
-					},
-				},
-			}
 
-			Eventually(func() error {
-				kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(context.Background(), kv, metav1.UpdateOptions{})
-				return err
-			}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+			_, err = patch.New(patch.WithAdd("/spec/template/spec/nodeSelector/kubernetes.io~1hostname", "does-not-exist")).GeneratePayload()
```

Comment:
I think this is wrong! Other than that no patch is being performed, with this you are going to patch `/spec/template/spec/nodeSelector/kubernetes.io~1hostname` which actually does not exist.
The patch that should be performed is the addition of the `kv.spec.customizedComponents`:
It should be something like this:
```
originalKv := util.GetCurrentKv(virtClient)
customizedComponents := v1.CustomizeComponents{
	Patches: []v1.CustomizeComponentsPatch{
		{
			ResourceName: virtHandler.deploymentName,
			ResourceType: "DaemonSet",
			Patch:        `{"spec":{"template":{"spec":{"nodeSelector":{"kubernetes.io/hostname":"does-not-exist"}}}}}`,
			Type:         v1.StrategicMergePatchType,
		},
	},
}
patchBytes, err := patch.New(patch.WithAdd("/spec/customizeComponents", customizedComponents)).GeneratePayload()
Expect(err).ToNot(HaveOccurred())
kv, err := virtClient.KubeVirt(originalKv.Namespace).Patch(context.Background(), originalKv.Name, types.JSONPatchType, patchBytes, metav1.PatchOptions{})
Expect(err).ToNot(HaveOccurred())
```
In this way, you are patching the KubeVirt resource, which will patch the virt-handler with the patch inside the customizedComponent.